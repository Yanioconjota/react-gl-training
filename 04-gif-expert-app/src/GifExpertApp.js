import { Col } from 'react-bootstrap';
import "./GifExpertApp.scss";
import { useState } from 'react';

import AddCategory from './components/AddCategory';

const GifExpertApp = () => {

  const [categories, setCategories] = useState([
    "One Punch Man",
    "Samurai X",
    "Dragon Ball Z",
  ]);

  /* const handleAdd = () => { 
    setCategories(newCategories => [...newCategories, `categoria-${Math.random()}`])
  }; */

  return (
    <Col xs={12} md={6}>
      <h2 className="text-info">GifExpertApp</h2>
      <hr />
      <AddCategory setCategories={setCategories} />
      <ol>
        {categories.map((category) => {
          return <li key={category}>{category}</li>;
        })}
      </ol>
    </Col>
  );

}

export default GifExpertApp;