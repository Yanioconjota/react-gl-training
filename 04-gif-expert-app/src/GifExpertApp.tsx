import { Col } from 'react-bootstrap';
import "./GifExpertApp.scss";
import PropTypes from 'prop-types';
import { useState } from 'react';

import AddCategory from './components/AddCategory';
import GifGrid from './components/GifGrid';

const GifExpertApp = () => {

  const [categories, setCategories] = useState([
    "Far Cry 6",
  ]);

  return (
    <Col xs={12}>
      <h2 className="text-info">GifExpertApp</h2>
      <hr />
      <AddCategory setCategories={setCategories} categories={categories} />
      <>
        <GifGrid key={categories} category={categories} />
      </>
    </Col>
  );
}

GifExpertApp.propTypes = {
  categories: PropTypes.arrayOf(PropTypes.string),
  setCategories: PropTypes.func
}

export default GifExpertApp;