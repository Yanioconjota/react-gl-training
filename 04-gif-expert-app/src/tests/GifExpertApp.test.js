import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import GifExpertApp from "../GifExpertApp";

describe('Pruebas con GifExpertApp', () => {
  test('el componente debe de mostrarse correctament', () => {
    const wrapper = shallow(<GifExpertApp defaultCategories={'Far Cry 6'} />);
    expect(wrapper).toMatchSnapshot();
  });
});