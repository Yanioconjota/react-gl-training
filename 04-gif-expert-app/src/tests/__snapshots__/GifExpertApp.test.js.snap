// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con GifExpertApp el componente debe de mostrarse correctament 1`] = `
<Col
  xs={12}
>
  <h2
    className="text-info"
  >
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    categories="Far Cry 6"
    setCategories={[Function]}
  />
  <GifGrid
    category="Far Cry 6"
  />
</Col>
`;
